### PR TITLE
Secrets manager support

### PR DESCRIPTION
**Pull request by siliconsenthil**

### CODE REVIEW

File: cloudlift/deployment/service_information_fetcher.py [hunk 1]
```diff
@@ -44,57 +49,25 @@ def init_stack_info(self):
 
     def get_current_version(self):
         commit_sha = self._fetch_current_task_definition_tag()
-        if commit_sha is None or commit_sha == 'dirty':
-            log_warning("Currently deployed tag could not be found or is dirty,\
-resetting to master")
-            commit_sha = "master"
-        return commit_sha
-
-    def log_ips(self):
```
Comment (siliconsenthil) [inline cloudlift/deployment/service_information_fetcher.py:53]: Dead code removal

File: cloudlift/deployment/service_information_fetcher.py [hunk 2]
```diff
@@ -7,29 +7,34 @@
 from cloudlift.deployment.ecs import DeployAction, EcsClient
 from cloudlift.config import get_region_for_environment
 
+
 class ServiceInformationFetcher(object):
     def __init__(self, name, environment):
         self.name = name
         self.environment = environment
         self.cluster_name = get_cluster_name(environment)
-        self.ecs_client = get_client_for('ecs', self.environment)
```
Comment (siliconsenthil) [inline cloudlift/deployment/service_information_fetcher.py:15]: Refactor: Moved them as local variables where needed. Makes testing easier and code lighter

File: cloudlift/deployment/service_template_generator.py [hunk 3]
```diff
@@ -174,24 +177,26 @@ def _add_service_alarms(self, svc):
...
+            "Secrets": [Secret(Name=name, ValueFrom=secrets_config[name]) for name in secrets_config],
             "Name": container_name(service_name),
             "Image": self.ecr_image_uri + ':' + self.current_version,
             "Essential": 'true',
             "LogConfiguration": log_config,
             "MemoryReservation": int(config['memory_reservation']),
             "Cpu": 0
         }
+        if secrets_name_prefix:
+            self.template.add_output(Output(service_name + "SecretsNamePrefix",
```
Comment (siliconsenthil) [inline cloudlift/deployment/service_template_generator.py:197]: Adding secret name prefix as an output to be used during deployment. The cfn output is preferred as it reflects the true state of the stack. The dynamoDB config can be outdated as there can be  failure between dynamoDB update and cfn apply

File: cloudlift/deployment/service_template_generator.py [hunk 4]
```diff
@@ -270,12 +275,36 @@ def _add_service(self, service_name, config):
                 ]
             )
         ))
+        # https://docs.aws.amazon.com/code-samples/latest/catalog/iam_policies-secretsmanager-asm-user-policy-grants-access-to-secret-by-name-with-wildcard.json.html
+        secrets_arn_regex = f"arn:aws:secretsmanager:{self.region}:{self.account_id}:secret:*-{self.env}-??????"
```
Comment (siliconsenthil) [inline cloudlift/deployment/service_template_generator.py:279]: taskExecutionRole is [needed](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_execution_IAM_role.html) to pull secrets when the container is started.

File: cloudlift/deployment/service_updater.py [hunk 5]
```diff
@@ -36,10 +34,10 @@ def __init__(self, name, environment, env_sample_file, timeout_seconds, version=
         self.build_args = build_args
         self.dockerfile = dockerfile
         self.working_dir = working_dir
+        self.service_info = ServiceInformationFetcher(self.name, self.environment).service_info
 
     def run(self):
         log_warning("Deploying to {self.region}".format(**locals()))
-        self.init_stack_info()
```
Comment (siliconsenthil) [inline cloudlift/deployment/service_updater.py:42]: The same logic was present in `ServiceInformationFetcher`. So removed that logic here and reusing ServiceInformationFetcher

File: test/test_cloudlift.py [hunk 6]
```diff
@@ -2,15 +2,17 @@
...
 import urllib3
 from mock import patch
+import json
 
 from cloudlift.config import ServiceConfiguration, VERSION
 from cloudlift.deployment.service_creator import ServiceCreator
 from cloudlift.deployment.service_updater import ServiceUpdater
+from pathlib import Path
 
+TEST_DIR = Path(__file__).resolve().parent
```
Comment (siliconsenthil) [inline test/test_cloudlift.py:15]: Using pathlib to avoid the dependency of the test being run from specific dir

**Review by aswinkarthik — CHANGES_REQUESTED** [review]

**Review by aswinkarthik — APPROVED** [review]

**Pull request merged**